core/snapshot_util.rs

```diff
@@ -153,40 +153,6 @@ fn data_error_to_panic(err: v8::DataError) -> ! {
   }
 }
 
-pub(crate) enum SnapshotOptions {
```

Comment:
I'd like to keep this, however it might be easier to split it into two enums. Let's rename it to `SnapshotMode` and give it enum values like `WillSnapshot` or `WillNotSnapshot`. 

For the `Load`/`CreateFromExisting` we can use the `Option<Snapshot>` as before where we are potentially carrying around a snapshot, and use `BindingsMode` instead of `used_snapshot`.

Once we have a chance to tackle cleanup of extensions and bindings, these will be more useful.

Reply:
Now `will_snapshot` is not kept as state in `JsRuntime`. Do we want to add the enum just to pass an arg to `new_inner()`? Not fond of it..

Reply:
I'm OK with limited-scope booleans that are clear like this -- if they aren't crossing between code units that's fine.